File: Web.UI/src/types/record.ts
export type RecordResult = {
    records: Record[],
    count: number
}

export type Record = {
    date: string,
    total: number,
    duration: Duration,
    rating: Rating,
    response: Response,
    tags: Tags
}

export type Duration = {
    agentsChattingDuration: number,
    duration: number,
    count: number
}

export type Rating = {
    good: number,
    bad: number,
    chats: number
}

export type Response = {
    responseTime: number,
    count: number
}

export type Tags = {
    [name: string]: number
}

File: Web.Core/Entities/Record.cs
namespace Web.Core.Entities;

public sealed class Record
{
    public required string Date { get; set; }
    public required int Total { get; init; }
    public required Continuance Duration { get; init; } = new();
    public required Rating Rating { get; init; } = new();
    public required Response Response { get; init; } = new();
    public required Dictionary<string, int> Tags { get; init; } = [];
}
